imjoy/__main__.py

```diff
@@ -57,7 +59,10 @@ def main():
         ).wait()
         if ret != 0:
             print("Failed to upgrade ImJoy Plugin Engine.")
-        from .imjoyPluginEngine import main
+
+        # reload to use the new version
+        importlib.reload(imjoy)
```

Comment:
To use it here, it seems we cannot reload if we don't import it.

Reply:
Will it reload the subpackages of `imjoy` too. We're only import the options here so that's the only thing we would need to reload, right?

Reply:
It doesn't seem work like that during my test. It seems if you import even a submodule, the whole package will be cached.
I could get the right version number if I set a different version number and allow it to upgrade from github.

So, I think import and reload resolves the entire package, not submodules, which make sense.

Reply:
It will import the package if you import eg `from imjoy.options import parse_cmd_line` but we don't have anything yet in the package here, so that should matter?

Reply:
I had to use `reload` to get to the new version even if we only import a submodule.